Tests/Pcap++Test/Tests/LiveDeviceTests.cpp

```diff
@@ -379,91 +380,188 @@ PTF_TEST_CASE(TestPcapLiveDeviceStatsMode)
+	PTF_ASSERT_GREATER_OR_EQUAL_THAN(std::system(iptablesAddInputDrop.c_str()), 0);
+	PTF_ASSERT_GREATER_OR_EQUAL_THAN(std::system(iptablesAddOutputDrop.c_str()), 0);
+
+	// recover the interface at the end
+	SystemCommandTeardown iptablesDeleteInputDrop("sudo iptables -D INPUT -i " + interfaceName + " -j DROP");
+	SystemCommandTeardown iptablesDeleteOutputDrop("sudo iptables -D OUTPUT -o " + interfaceName + " -j DROP");
+
+	// open device
+	pcpp::PcapLiveDevice::DeviceConfiguration newConfig;
+	newConfig.usePoll = true;
+
+	PTF_ASSERT_TRUE(liveDev->open(newConfig));
+	DeviceTeardown devTeardown(liveDev);
+
+	int packetCount = 0;
+	PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeTimeout, &packetCount, 2), -1);
+	PTF_ASSERT_EQUAL(packetCount, 0);
 
 	liveDev->close();
-	PTF_ASSERT_FALSE(liveDev->isOpened());
+#else
+	PTF_SKIP_TEST("This test can not run in Windows environment");
+#endif
+} // TestPcapLiveDeviceBlockingMode
 
-	// a negative test
-	pcpp::Logger::getInstance().suppressLogs();
-	PTF_ASSERT_FALSE(liveDev->startCapture(packetArrives, &packetCount));
-	pcpp::Logger::getInstance().enableLogs();
+PTF_TEST_CASE(TestPcapLiveDeviceBlockingModePollNotTimeout)
+{
+#if !defined(_WIN32)
+	pcpp::PcapLiveDevice* liveDev = pcpp::PcapLiveDeviceList::getInstance().getPcapLiveDeviceByIp(PcapTestGlobalArgs.ipToSendReceivePackets.c_str());
+	auto interfaceName = liveDev->getName();
+
+	// drop all packets on the interface
+	auto iptablesAddInputDrop = "sudo iptables -A INPUT -i " + interfaceName + " -j DROP";
+	auto iptablesAddOutputDrop = "sudo iptables -A OUTPUT -o " + interfaceName + " -j DROP";
+	PTF_ASSERT_GREATER_OR_EQUAL_THAN(std::system(iptablesAddInputDrop.c_str()), 0);
+	PTF_ASSERT_GREATER_OR_EQUAL_THAN(std::system(iptablesAddOutputDrop.c_str()), 0);
+
+	// recover the interface at the end
+	SystemCommandTeardown iptablesDeleteInputDrop("sudo iptables -D INPUT -i " + interfaceName + " -j DROP");
+	SystemCommandTeardown iptablesDeleteOutputDrop("sudo iptables -D OUTPUT -o " + interfaceName + " -j DROP");
+
+	// open device
+	pcpp::PcapLiveDevice::DeviceConfiguration newConfig;
+	newConfig.usePoll = false; // explicitly mention not using poll
+
+	PTF_ASSERT_TRUE(liveDev->open(newConfig));
+	DeviceTeardown devTeardown(liveDev);
+
+	int packetCount = 0;
+
+	auto func = [&]() -> int
+	{
+		return liveDev->startCaptureBlockingMode(packetArrivesBlockingModeTimeout, &packetCount, 2);
+	};
+
+	// test it not timeout after 2 seconds
+	std::packaged_task<int()> task(func);
+
+    std::future<int> future = task.get_future();
+
+    // Start a new thread to execute `startCaptureBlockingMode` and to test it timeout
+    std::thread thread(std::move(task));
+
+    // Wait for the function to finish or the timeout to occur
+    auto status = future.wait_for(std::chrono::milliseconds(2100)); // a little bit more than 2 seconds
+
+	// the function doesn't timeout
+	PTF_ASSERT_TRUE(status == std::future_status::timeout);
```

Comment:
this line tells us that the function has a timeout.

Reply:
Does it tell if the timeout was because of `poll` timeout or because the while loop finished after timeout? 🤔

Reply:
please notice that, if `poll` timeout, it should timeout at right after 2000ms, and I set future timeout at 2100ms. If you have concern, I can also set timeout of future to be like 3000ms.

Reply:
ops, I think you misunderstood.
This test doesn't use poll.
`TestPcapLiveDeviceBlockingModeNotTimeoutWithoutPoll` is explaining, the original implementation can not successfully timeout. So I use `future` to monitor it.
If the original function is correct,  it should timeout itself. `future` will not get `std::future_status::timeout`.

Reply:
I still don't understand... how does `future` ensure the timeout is because of `poll` timeout and not because of something else?

Reply:
in this test, we don **NOT** use `poll`; we are using the original implementation.

Recall that the PR is trying to resolve that the `pcap_dispatch` call will never break if there is no packets coming, leading to `while(!timeout)` never being checked(, hence it get stuck)

so, in this test, we disable the interface, so there is no packets coming, and by using the original implementation, `pcap_dispatch` will block and wait for packets. We use another thread (by `future`) to run the blocking call, and the main thread will monitor if the `startCaptureBlockingMode` can succecully timeout.

**if the original implementation is correct**, `startCaptureBlockingMode` will timeout right after 2000ms, and the `futre` will get `std::future_status::ready`. However, **if `startCaptureBlockingMode` is still blocking**, after the `future`  gets timeout, say 3000ms, `future` will get  `std::future_status::timeout`.

Once `future` gets `std::future_status::timeout`, we know that `startCaptureBlockingMode` is still blocking and doen't break, so we enble the interface, and there are packets coming, and `startCaptureBlockingMode` can go to the next loop, now its `while(!timeout)` is execute, and the functoin call finally finished. If we correctly run `thread.join()`, it means that the `startCaptureBlockingMode` thread correctly closes (otherwise it means the thread is still blocking)

Reply:
Thanks for the explanation @tigercosmos ! I thing I understand it better now.
However, I'm not sure what the test is trying to test - the whole point of this PR is to use `poll` because `pcap_dispatch` tends to get stuck if there are no incoming packets. I'd expect the test will run the scenario I mentioned:
- we use `poll`
- we block the network traffic
- in the original implementation we'd expect `future` to get `std::future_status::timeout` because `pcap_dispatch` got stuck waiting for packets that don't arrive
- however, in the new implementation `poll` times out so we don't get stuck and we always get `std::future_status::ready`

What am I missing here?